tests/admin_inlines/tests.py

```diff
@@ -226,6 +227,113 @@ def test_inline_hidden_field_no_column(self):
+        parent1_ct = ContentType.objects.get_for_model(Parent1)
+        child1_ct = ContentType.objects.get_for_model(Child1)
+        permission = Permission.objects.get(codename='view_parent1', content_type=parent1_ct)
+        user.user_permissions.add(permission)
+        permission = Permission.objects.get(codename='view_child1', content_type=child1_ct)
+        user.user_permissions.add(permission)
+        self.client.login(username="user", password="pwd")
+        response = self.client.get(reverse('admin:admin_inlines_parent1_change', args=(parent.pk,)))
+        self.assertInHTML('<th class="column-position hidden">Position</th>', response.rendered_content)
+        self.assertInHTML('<td class="field-position hidden"><p>0</p></td>', response.rendered_content)
+        self.assertInHTML('<td class="field-position hidden"><p>1</p></td>', response.rendered_content)
+
+    def test_stacked_inline_hidden_field_with_view_only_permissions(self):
+        """#31867 -- Make sure content of hidden field is not visible in stacked inline when user has view-only
+        permission"""
+        parent = Parent2.objects.create(name='a')
+        Child2.objects.create(name='b', position='0', parent=parent)
+        Child2.objects.create(name='c', position='1', parent=parent)
+        user = User.objects.create_user(username="user", password="pwd", is_staff=True)
+        parent2_ct = ContentType.objects.get_for_model(Parent2)
+        child2_ct = ContentType.objects.get_for_model(Child2)
+        permission = Permission.objects.get(codename='view_parent2', content_type=parent2_ct)
+        user.user_permissions.add(permission)
+        permission = Permission.objects.get(codename='view_child2', content_type=child2_ct)
+        user.user_permissions.add(permission)
+        self.client.login(username="user", password="pwd")
+        response = self.client.get(reverse('admin:admin_inlines_parent2_change', args=(parent.pk,)))
+        # Following test ensure that the whole line containing name + position fields is not hidden
+        self.assertContains(response, '<div class="form-row field-name field-position">')
+        # Following test ensure that the div containing the position field is hidden
+        self.assertInHTML(
+            '<div class="fieldBox field-position hidden"><label class="inline">Position:</label>'
+            '<div class="readonly">0</div></div>',
+            response.rendered_content
+        )
+        self.assertInHTML(
+            '<div class="fieldBox field-position hidden"><label class="inline">Position:</label>'
+            '<div class="readonly">1</div></div>',
+            response.rendered_content
+        )
+
+    def test_stacked_inline_single_hidden_field_in_line_with_view_only_permissions(self):
+        """#31867 -- Make sure content of hidden field is not visible in stacked inline when user has view-only
+        permission and field is the only field on one line"""
```

Comment:
Please chop the ticket number. Also, to add more explanation, docstrings should state the expected behavior and omit prefixes like "Make sure" since all tests ensure things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

This comment applies to all new docstrings.